src/basilisp/test.lpy

```diff
@@ -67,6 +73,42 @@
   [_ & fixtures]
   (alter-meta! *ns* assoc ::once-fixtures fixtures))
 
+(defmacro with-fixtures
+  "Wrap the ``body`` in the ``fixtures`` in the given order. Handle
+  setup and teardown for each of the ``fixtures``."
+  [fixtures & body]
+  (assert (vector? fixtures) "Expected a literal vector of fixtures")
+  (let [result (gensym "result")]
+    (reduce (fn [form fixture]
+              `(let [~result (~fixture)]
+                 (if (generator? ~result)
+                   (try
+                     (python/next ~result)
+                     ~form
+                     (finally
+                       (try
+                         (python/next ~result)
+                         (catch python/StopIteration ~'_ nil))))
+                   ~form)))
+            `(do ~@body)
+            (reverse fixtures))))
+
+(defmacro compose-fixtures
+  "Compose any number of ``fixtures``, in order, creating a new fixture
+  that combines their behavior. Always returns a valid fixture
+  function, even if no fixtures are given."
+  [& fixtures]
+  `(fn [] (with-fixtures [~@fixtures] (yield))))
+
+(defn join-fixtures
+  "Composes a collection of ``fixtures``, in order.  Always returns a valid
+  fixture function, even if the collection is empty.
+  Prefer :lpy:macro`compose-fixtures` if fixtures are known at compile time."
```

Comment:
```suggestion
  Prefer :lpy:fn:`compose-fixtures` if fixtures are known at compile time."
```

I'm not sure if it was a helpful or harmful decision on my part, but the Sphinx autodocumenting plugin doesn't distinguish between macros and functions, so you can just use `:lpy:fn:`.